### PR TITLE
Official symbols

### PR DESCRIPTION
**Pull request by SashaIr**
New graphics.

### CODE REVIEW

File: public/js/src/game/ui/drawPip.ts [hunk 1]
```diff
@@ -1,164 +1,762 @@
...
-    ctx.moveTo(points[0][0], points[0][1]);
-    ctx.quadraticCurveTo(interps[0][0], interps[0][1], points[1][0], points[1][1]);
-    ctx.quadraticCurveTo(interps[1][0], interps[1][1], points[2][0], points[2][1]);
-    ctx.quadraticCurveTo(interps[2][0], interps[2][1], points[3][0], points[3][1]);
-    ctx.quadraticCurveTo(interps[3][0], interps[3][1], points[0][0], points[0][1]);
+const shapeFunctions: Map<string, (ctx: CanvasRenderingContext2D, colors?: Array<string>) => Array<any>> = new Map();
+const sf = 3; // Scale factor
+
+shapeFunctions.set('hanabi_red', (ctx: CanvasRenderingContext2D) => {
+
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: error

File: public/js/src/game/ui/drawPip.ts [hunk 2]
```diff
@@ -1,164 +1,762 @@
...
+    ctx.bezierCurveTo(70.205570, 87.912670, 64.799917, 59.401233, 38.201088, 59.405974);
+    ctx.scale(1/sf, 1/sf);
     ctx.closePath();
 
     return [];
 });
 
-shapeFunctions.set('club', (ctx: CanvasRenderingContext2D) => {
+shapeFunctions.set('hanabi_yellow', (ctx: CanvasRenderingContext2D) => {
+
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: error

File: public/js/src/game/ui/drawPip.ts [hunk 3]
```diff
@@ -1,164 +1,762 @@
...
-        rot += step;
-
-        const x2 = cx + (Math.cos(rot) * innerRadius);
-        const y2 = cy + (Math.sin(rot) * innerRadius);
-        ctx.lineTo(x2, y2);
-        rot += step;
-    }
-    ctx.lineTo(cx, cy - outerRadius);
+shapeFunctions.set('hanabi_green', (ctx: CanvasRenderingContext2D) => {
+
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: error

File: public/js/src/game/ui/drawPip.ts [hunk 4]
```diff
@@ -1,164 +1,762 @@
...
+    ctx.bezierCurveTo(58.374671, 61.846624, 59.841692, 60.661853, 56.910015, 58.292302);
+    ctx.scale(1/sf, 1/sf);
     ctx.closePath();
 
     return [];
 });
 
-shapeFunctions.set('heart', (ctx: CanvasRenderingContext2D) => {
+shapeFunctions.set('hanabi_blue', (ctx: CanvasRenderingContext2D) => {
+
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: error

File: public/js/src/game/ui/drawPip.ts [hunk 5]
```diff
@@ -1,164 +1,762 @@
...
+    ctx.bezierCurveTo(120.546360, 94.541477, 116.388350, 89.549157, 115.298220, 63.130560);
+    ctx.scale(1/sf, 1/sf);
     ctx.closePath();
 
     return [];
 });
 
-shapeFunctions.set('crescent', (ctx: CanvasRenderingContext2D) => {
+shapeFunctions.set('hanabi_white', (ctx: CanvasRenderingContext2D) => {
+
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: error

File: public/js/src/game/ui/drawPip.ts [hunk 6]
```diff
@@ -1,164 +1,762 @@
...
+    ctx.scale(4,4);
 
     return [];
 });
 
-shapeFunctions.set('flower', (ctx: CanvasRenderingContext2D) => {
-    ctx.font = '190px Verdana';
-    return ['✿', -10, 155];
+shapeFunctions.set('hanabi_alt_white', (ctx: CanvasRenderingContext2D) => {
+
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: error

File: public/js/src/game/ui/drawPip.ts [hunk 7]
```diff
@@ -1,164 +1,762 @@
...
+    ctx.bezierCurveTo(23.170303, 14.246451, 23.429261, 14.165214, 23.660156, 14.164062);
+    ctx.scale(1/sf, 1/sf);
+    ctx.closePath();
+
+    return [];
 });
 
-shapeFunctions.set('spade', (ctx: CanvasRenderingContext2D) => {
+shapeFunctions.set('hanabi_alt_blue', (ctx: CanvasRenderingContext2D) => {
+
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: error

File: public/js/src/game/ui/drawPip.ts [hunk 8]
```diff
@@ -1,164 +1,762 @@
...
     ctx.closePath();
 
     return [];
 });
 
-shapeFunctions.set('number_sign', (ctx: CanvasRenderingContext2D) => {
-    ctx.font = '190px Verdana';
-    return ['#', -3, 170];
+shapeFunctions.set('white_moon', (ctx: CanvasRenderingContext2D) => {
+
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: error

File: public/js/src/game/ui/drawPip.ts [hunk 9]
```diff
@@ -1,164 +1,762 @@
...
+    ctx.scale(1/sf, 1/sf);
+    ctx.closePath();
+    ctx.scale(4, 4);
+
+    return [];
 });
 
-shapeFunctions.set('circle', (ctx: CanvasRenderingContext2D) => {
+shapeFunctions.set('pink_hashtag', (ctx: CanvasRenderingContext2D) => {
+
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: error

File: public/js/src/game/ui/drawPip.ts [hunk 10]
```diff
@@ -1,164 +1,762 @@
...
+    ctx.scale(3,3);
 
     return [];
 });
 
-shapeFunctions.set('tensor_symbol', (ctx: CanvasRenderingContext2D) => {
-    ctx.font = '170px Verdana';
-    return ['⊗', -9, 150];
+shapeFunctions.set('brown_splat', (ctx: CanvasRenderingContext2D) => {
+
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: error

File: public/js/src/game/ui/drawPip.ts [hunk 11]
```diff
@@ -1,164 +1,762 @@
...
+    ctx.scale(3,3);
+
+    return [];
 });
 
-shapeFunctions.set('infinity', (ctx: CanvasRenderingContext2D) => {
-    ctx.font = '175px Verdana';
-    return ['∞', -10, 155];
+shapeFunctions.set('empty_set', (ctx: CanvasRenderingContext2D) => {
+
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: error

File: public/js/src/game/ui/drawPip.ts [hunk 12]
```diff
@@ -1,164 +1,762 @@
...
+    ctx.scale(3.5,3.5);
+
+    return [];
 });
 
-shapeFunctions.set('null_symbol', (ctx: CanvasRenderingContext2D) => {
-    ctx.font = '210px Verdana';
-    return ['∅', 10, 165];
+shapeFunctions.set('infinity', (ctx: CanvasRenderingContext2D, colors?: Array<string>) => {
+
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: error

File: public/js/src/game/ui/drawPip.ts [hunk 13]
```diff
@@ -1,164 +1,762 @@
...
+        ctx.fillStyle = gradient;
+        ctx.fill();
+    }
+
+    return [];
 });
 
-shapeFunctions.set('rainbow', (ctx: CanvasRenderingContext2D) => {
+shapeFunctions.set('rainbow', (ctx: CanvasRenderingContext2D, colors?: Array<string>) => {
+
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: error

File: public/js/src/game/ui/drawPip.ts [hunk 14]
```diff
@@ -1,164 +1,762 @@
...
+    ctx.bezierCurveTo(74.463876, 120.864520, 73.761523, 120.627410, 73.172033, 120.342210);
+    ctx.bezierCurveTo(72.711854, 115.080550, 75.752370, 115.504430, 77.638733, 116.418060);
+    ctx.bezierCurveTo(77.955104, 120.035450, 76.616822, 120.965250, 75.136667, 120.896220);
+    ctx.transform(1/2.128652, 0.000000, 0.000000, 1/2.476848, 0, 0);
+    ctx.scale(1/1.2, 1/1.2);
+    ctx.scale(1/sf, 1/sf);
+    ctx.closePath();
+
+    if (colors) {
+    // When multiple colors are passed, create a gradient.
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: indent

File: public/js/src/game/ui/drawPip.ts [hunk 15]
```diff
@@ -1,164 +1,762 @@
...
+    ctx.bezierCurveTo(113.304710, 121.143780, 115.386250, 116.996300, 118.045660, 114.266600);
+    ctx.bezierCurveTo(118.602190, 113.695410, 119.833220, 112.584330, 119.908670, 112.585320);
+    ctx.bezierCurveTo(119.925970, 112.585560, 120.285310, 112.748660, 120.704910, 112.947790);
+    if (colors) {
+      ctx.fillStyle = colors[1];
+      ctx.fill();
+    }
+    ctx.closePath();
+
+// 2515
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: indent

File: public/js/src/game/ui/drawPip.ts [hunk 16]
```diff
@@ -1,164 +1,762 @@
...
+    ctx.bezierCurveTo(105.074040, 97.360785, 105.669280, 96.739035, 107.188930, 95.723645);
+    ctx.bezierCurveTo(110.261110, 93.670755, 114.325170, 91.716045, 116.863790, 91.070105);
+    ctx.bezierCurveTo(117.650790, 90.869705, 118.613070, 90.808665, 118.989640, 90.934275);
+    if (colors) {
+      ctx.fillStyle = colors[4];
+      ctx.fill();
+    }
+    ctx.closePath();
+
+// 2517
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: indent

File: public/js/src/game/ui/drawPip.ts [hunk 17]
```diff
@@ -1,164 +1,762 @@
...
+    ctx.bezierCurveTo(102.158430, 107.709460, 102.343840, 107.321480, 102.568270, 107.129110);
+    ctx.bezierCurveTo(102.855560, 106.882850, 104.280190, 106.522360, 105.763100, 106.320830);
+    ctx.bezierCurveTo(106.706170, 106.192410, 108.973620, 106.132430, 109.895770, 106.211460);
+    if (colors) {
+      ctx.fillStyle = colors[0];
+      ctx.fill();
+    }
+    ctx.closePath();
+
+    //2521
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: what does this comment mean

File: public/js/src/game/ui/drawPip.ts [hunk 18]
```diff
@@ -1,164 +1,762 @@
...
+    ctx.bezierCurveTo(128.279810, 130.553510, 127.113200, 129.549050, 126.993320, 129.446690);
+    ctx.bezierCurveTo(123.648090, 126.259020, 120.822940, 116.510980, 121.950910, 113.876780);
+    ctx.bezierCurveTo(122.064230, 113.617110, 122.150180, 113.382000, 122.214290, 113.155620);
+    ctx.bezierCurveTo(123.808760, 118.432660, 129.048120, 123.864510, 134.649160, 125.744850);
+    ctx.bezierCurveTo(134.766640, 125.784350, 134.895210, 125.804050, 135.025800, 125.804050);
+    ctx.bezierCurveTo(137.763210, 125.804050, 142.174240, 116.278160, 143.077430, 113.249640);
+    ctx.bezierCurveTo(143.372690, 112.256070, 143.379280, 111.584970, 143.094730, 111.197170);
     ctx.closePath();
 
+    ctx.scale(1/sf, 1/sf);
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: space your operators

File: public/js/src/game/ui/drawPip.ts (hunk 17 above)
Comment (SashaIr) [reply public/js/src/game/ui/drawPip.ts]: It's the name of the path in my svg file, is there for my own sanity, but I can change it to something meaningful.

File: misc/keldon/ui.js [hunk 19]
```diff
@@ -1277,10 +1277,10 @@ var show_clue_match = function(target, clue, show_neg) {
 };
 
 var suit_colors = [
-	"#0044cc",
-	"#00cc00",
-	"#ccaa22",
-	"#aa0000",
+	"#0037ff",
```
[the remark this answers is not in the document]
Comment (SashaIr) [reply misc/keldon/ui.js]: That was a find and replace.

File: .gitignore [hunk 20]
```diff
@@ -37,3 +37,7 @@ hanabi-live~
 
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
+
+# Konva modules
+node_modules
```
[the remark this answers is not in the document]
Comment (SashaIr) [reply .gitignore]: Fixed.

File: public/js/src/game/ui/HanabiCardInit.ts [hunk 21]
```diff
@@ -111,13 +111,11 @@ export function pips(this: HanabiCard) {
...
         };
         // Transform polar to Cartesian coordinates
-        // The magic number added to the offset is needed to center things properly
-        // We don't know why it's needed; perhaps something to do with the shape functions
         const offsetBase = CARD_W * 0.7;
         const offsetTrig = ((-i / suits.length) + 0.25) * Math.PI * 2;
         const offset = {
-            x: Math.floor((offsetBase * Math.cos(offsetTrig)) + (CARD_W * 0.25)),
-            y: Math.floor((offsetBase * Math.sin(offsetTrig)) + (CARD_W * 0.3)),
+            x: Math.floor((offsetBase * Math.cos(offsetTrig))),
```
[the remark this answers is not in the document]
Comment (SashaIr) [reply public/js/src/game/ui/HanabiCardInit.ts]: Fixed.

File: public/js/src/game/ui/drawCards.ts [hunk 22]
```diff
@@ -81,7 +81,16 @@ export default (variant: Variant, colorblindUI: boolean) => {
                 ctx.font = `bold ${fontSize}pt Arial`;
 
                 // Draw the rank on the top left
-                drawText(ctx, textYPos, rankLabel);
+                if (colorblindUI) {
+                    drawText(ctx, textYPos, rankLabel);
+                } else {
+    		        ctx.save();
```
[the remark this answers is not in the document]
Comment (SashaIr) [reply public/js/src/game/ui/drawCards.ts]: Fixed.

File: public/js/src/game/ui/drawPip.ts [hunk 23]
```diff
@@ -1,164 +1,661 @@
...
+    ctx.scale(3, 3);
 
-shapeFunctions.set('number_sign', (ctx: CanvasRenderingContext2D) => {
-    ctx.font = '190px Verdana';
-    return ['#', -3, 170];
+    return [];
 });
 
-shapeFunctions.set('circle', (ctx: CanvasRenderingContext2D) => {
+shapeFunctions.set('brown_splat', (ctx: CanvasRenderingContext2D) => {
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: rename this?

File: public/js/src/game/ui/drawPip.ts [hunk 24]
```diff
@@ -168,40 +665,31 @@ export default (
     shadow: boolean,
     deckBack: boolean,
 ) => {
-    // Each suit has a shape defined in the "suits.json" file (as the "pip" property)
+    // Each suit has a shape defined in the 'suits.json' file (as the 'pip' property)
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: these should be double quotes

File: public/js/src/game/ui/drawPip.ts [hunk 25]
```diff
@@ -168,40 +665,31 @@ export default (
...
     const shapeFunction = shapeFunctions.get(suit.pip);
     if (!shapeFunction) {
-        throw new Error(`Failed to find the shape function for pip "${suit.pip}".`);
+        throw new Error(`Failed to find the shape function for pip '${suit.pip}'.`);
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: these should be double quotes

File: public/js/src/game/ui/drawPip.ts [hunk 26]
```diff
@@ -168,40 +665,31 @@ export default (
...
-        // Rainbow and omni cards have a gradient fill
-        // Generate a radial gradient that is evenly distributed between its component colors
-        const gradient = ctx.createRadialGradient(75, 150, 25, 75, 150, 75);
-        for (let i = 0; i < suit.fillColors.length; ++i) {
-            gradient.addColorStop(i / (suit.fillColors.length - 1), suit.fillColors[i]);
-        }
-        ctx.fillStyle = gradient;
+        // Rainbow and omni cards have a multiple color fill
+        // which is passed as an array to the drawing function
+        // The drawing function will handle the filling.
```
Comment (Zamiell) [inline public/js/src/game/ui/drawPip.ts]: no period here

**Comment by Zamiell** [comment]
1) still need the notes.txt documentation + svg images
2) can we make Cocoa Rainbow look different from Muddy Rainbow somehow? (and Chiaroscuro Pink different from Light Pink)

**Comment by Zamiell** [comment]
and lets remember to include the URL of http://tecs.g2.xrea.com/hyouki/kana/kana.html in the notes

**Comment by SashaIr** [comment]
_Muddy Rainbow_ and _Cocoa Rainbow_ are already different. I suggest to rename _Chiaroscuro Pink_ to _Gray Pink_ (alternatives welcome) so it makes sense to use a less saturated colour, and also the name of the suit is more descriptive of what the suit is.

**Comment by Zamiell** [comment]
> rename Chiaroscuro Pink to Gray Pink

fine with me

File: docs/VARIANTS.md [hunk 27]
```diff
@@ -22,7 +22,7 @@ One of each card per suit (e.g. 5 in total):
 | ------------------ | ---------------- | -------------- | --------------- |
 | No rank clues      | Dark Null        | Dark Brown     | Cocoa Rainbow   |
 | Own rank clue      | Gray             | Black          | Dark Rainbow    |
-| All rank clues     | Chiaroscuro Pink | Dark Pink      | Dark Omni       |
+| All rank clues     | Gray Pink | Dark Pink      | Dark Omni       |
```
Comment (Zamiell) [inline docs/VARIANTS.md]: oops, add a few spaces here so that the columns line up again

Comment (SashaIr) [reply docs/VARIANTS.md]: Whoops, sorry.

**Comment by Zamiell** [comment]
that was weird, gui refused to merge, but there were no problems or conflicts when i did it from command line

**Pull request merged**